api/cpp/include/slint_platform.h

```diff
@@ -157,6 +157,49 @@ class WindowAdapter
     }
 };
 
+/// An event that is passed to the Platform::invoke_from_event_loop function and needs to be invoked
+/// in the event loop.
+class PlatformEvent
+{
+    cbindgen_private::PlatformEventOpaque inner { nullptr, nullptr };
+    friend class Platform;
+
+    explicit PlatformEvent(cbindgen_private::PlatformEventOpaque inner) : inner(inner) { }
+
+public:
+    ~PlatformEvent()
+    {
+        if (inner._0) {
+            auto tmp = inner;
+            inner = { nullptr, nullptr };
```

Comment:
Maybe this could also be written as

```cpp
auto tmp = std::exchange(inner, { nullptr, nullptr });
```

(not sure it compiles though)

Reply:
I did not know about std::exchange, thanks!